---
title: "18S significantly changing community"
author: "Sven"
date: "2/25/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
##Notes:
It is best to go through each dataset (five fjords/horizontal Long Sound community/ Vertical Long Sound community) one by one. So following the order of this file from start to finish.

#Set up
```{r Load packages}
library(vegan)
library(phyloseq)
library(ggplot2)
library(dplyr)
library(microbiome)
library(scales)
```
##Import files:
```{r Set path}

uzdir <- "/Users/sven/Desktop/Project_R_analysis/"
otutable_biom_file <- paste("/Users/sven/Desktop/Project_R_analysis/", "Merged_otu_table_json.biom", sep = "")
map_file <- paste("/Users/sven/Desktop/Project_R_analysis/Sorted_out/", "Modified_mapping_file_mod_with_names_and_variables.txt", sep = "")


```
```{r Create phyloseq object}
# Now import the .biom-formatted otu_table-tax_table file.

biom_otu_tax <- import_biom(otutable_biom_file)

# Add sample data to the dataset using merge
bmsd <- import_qiime_sample_data(map_file)
class(bmsd)
dim(bmsd)
sample_data(bmsd)
sample_variables(bmsd)
biom_otu_tax


#Merge into phyloseq format
Fiord_phyloseq <- merge_phyloseq(biom_otu_tax, bmsd)
Fiord_phyloseq
sample_sums(Fiord_phyloseq)

#Save original phyloseq file
Fiord_phyloseq_v = Fiord_phyloseq

#Fix Misslabel
Fiord_phyloseq@sam_data$Sample_Depth[Fiord_phyloseq@sam_data$Sample_Depth == 1] = 0
#Ensure sample depth is seen as a factor for ease of use
sample_data(Fiord_phyloseq)$Sample_Depth<-as.factor(sample_data(Fiord_phyloseq)$Sample_Depth)
```
###Create average result for multiple rarefaction by transforming data using (divide by 10) and check counts per sample
```{r Create average result for multiple rarefaction by transforming data using (divide by 10), results='markup'}
Fiord_phyloseq = transform_sample_counts(Fiord_phyloseq, function(x) x/10)
sample_sums(Fiord_phyloseq)
```
#### Round and confirm count number
```{r Round and confirm count number, results='markup'}
Fiord_phyloseq = transform_sample_counts(Fiord_phyloseq, round)
sample_sums(Fiord_phyloseq)
Fiord_phyloseq = prune_samples(sample_sums(Fiord_phyloseq)>=1, Fiord_phyloseq)
sample_sums(Fiord_phyloseq)
```
Check that all OTUs have representative counts  
For here taxa = OTU  
__Commands interpretation:__  
_Total number of taxa in dataset:_ sum(taxa_sums(Fiord_phyloseq) > 0)   

_Any taxa with no hits:_ any(taxa_sums(Fiord_phyloseq)== 0)
```{r identify taxa with only zeros, results='markup', echo=TRUE}
sum(taxa_sums(Fiord_phyloseq) > 0)
any(taxa_sums(Fiord_phyloseq)== 0)
sum(taxa_sums(Fiord_phyloseq) == 0)
any(taxa_sums(Fiord_phyloseq) > 1)
sum(taxa_sums(Fiord_phyloseq) > 1)
any(taxa_sums(Fiord_phyloseq) < 1)
sum(taxa_sums(Fiord_phyloseq) < 1)
```
####Prune taxa with less than 1 count and check taxa numbers again
```{r  Save original file and create new file with only present (no zeroes) taxa, results='markup', echo=TRUE}

#Create new file with only present (no zeroes) taxa

Fiord_phyloseq = prune_taxa(taxa_sums(Fiord_phyloseq) > 1, Fiord_phyloseq)
any(sample_sums(Fiord_phyloseq) == 0)
any(sample_sums(Fiord_phyloseq) > 0)
sum(taxa_sums(Fiord_phyloseq) > 0)
any(sample_sums(Fiord_phyloseq) < 1)
sum(taxa_sums(Fiord_phyloseq) < 1)
```
###Compare sequences per sample or OTU
```{r Compare sequences per sample or OTU}
readsumsdf = data.frame(nreads = sort(taxa_sums(Fiord_phyloseq),TRUE), sorted = 1:ntaxa(Fiord_phyloseq), type = "OTU")
readsumsdf = rbind(readsumsdf,data.frame(nreads = sort(sample_sums(Fiord_phyloseq),TRUE),sorted = 1:nsamples(Fiord_phyloseq), type = "Samples"))

title = "Total number of reads"

p = ggplot(readsumsdf, aes(x = sorted, y = nreads)) +
  geom_bar(stat = "identity")

p + ggtitle(title) +
  scale_y_log10() +
  facet_wrap(~type, 1, scales = "free")
```
```{r What sample variables exist in the phyloseq data}
sample_variables(Fiord_phyloseq)
```
```{r Attached OTU ID}
tax_table(Fiord_phyloseq) <- cbind(tax_table(Fiord_phyloseq), OTU=taxa_names(Fiord_phyloseq))
```
```{r Rename Ranks}
colnames(tax_table(Fiord_phyloseq)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species", "OTU")
tax_table(Fiord_phyloseq) =gsub("D_0__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_1__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_2__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_3__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_4__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_5__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_6__", "", tax_table(Fiord_phyloseq))

```

##Modify files
```{r Add physicochemical parameters to mapping file}

Modified_mapping_file = read.csv("~/Desktop/Modified_mapping_file_18S.csv", row.names = 1)

sample_data(Fiord_phyloseq) = Modified_mapping_file

```
##Subset phyloseq objects
```{r Subset}
#Subset to in/out
Fiord_phyloseq_in_out = subset_samples(Fiord_phyloseq, !Sample_type == "Transverse")

#Subset horizontal community
Fiord_phyloseq_Horonly = subset_samples(Fiord_phyloseq, Sample_type == "Transverse" & !Sample_Depth == "40" & !Sample_Depth == "100" & !Sample_Depth == "200" & !Sample_Depth == "360")

#Subset vertical community
Fiord_phyloseq_Vert = subset_samples(Fiord_phyloseq, !Sample_type == "in" & !Sample_type == "out")
Fiord_phyloseq_Vert = subset_samples(Fiord_phyloseq_Vert, Sample_Site == "L5")
```
##Phylum colour list for consistency
```{r Phylum colour list}
Phylum_colour_list = c(Acidobacteria = "red",
                       Actinobacteria = "green",
                       "Alphaproteobacteria" = "burlywood1",
                       "Alveolata" = "blueviolet",
                       Archaeplastida = "#F8766D",
                       Bacteroidetes = "steelblue",  
                       Chlamydiae = "navy", 
                       Chloroflexi = "grey",
                       Cyanobacteria = "purple", 
                       "Deltaproteobacteria" = "orange",
                       "Discoba" = "darkslategray1",
                       Euryarchaeota = "aquamarine", 
                       "Epsilonproteobacteria" = "darkslateblue",
                       "Excavata" = "#7CAE00",
                       "Gammaproteobacteria" = "dimgray",
                       "Holozoa" = "forestgreen",
                       "Marine Group I" = "gainsboro",
                       Marinimicrobia = "darkseagreen1",
                       Opisthokonta = "magenta", 
                       Parcubacteria = "chocolate2", 
                       Planctomycetes = "wheat", 
                       Proteobacteria = "olivedrab",
                       "Rhizaria" = "goldenrod",
                       Thaumarchaeota = "firebrick", 
                       SAR = "darkred", 
                       "Stramenopiles" = "darkorchid1",
                       "Thermoplasmata" = "tan",
                       Verrucomicrobia = "gold4", 
                       "Rare Taxa (<1%)" = "black")
```
#Exact test function
```{r Exact test code}

#check open packages
(.packages())
##Close all but phyloseq
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()


library("phyloseq")
packageVersion("phyloseq")
library("edgeR")
packageVersion("edgeR")
library(phyloseq)
library(ggplot2)
library(plyr)
library(scales)
library(reshape)
library(RColorBrewer)
library(grid)
library(empiricalFDR.DESeq2)
library("DESeq2")
library(dplyr)
library(Rmisc)


#' Convert phyloseq OTU count data into DGEList for edgeR package
#' 
#' Further details.
#' 
#' @param physeq (Required).  A \code{\link{phyloseq-class}} or
#'  an \code{\link{otu_table-class}} object. 
#'  The latter is only appropriate if \code{group} argument is also a 
#'  vector or factor with length equal to \code{nsamples(physeq)}.
#'  
#' @param group (Required). A character vector or factor giving the experimental
#'  group/condition for each sample/library. Alternatively, you may provide
#'  the name of a sample variable. This name should be among the output of
#'  \code{sample_variables(physeq)}, in which case
#'  \code{get_variable(physeq, group)} would return either a character vector or factor.
#'  This is passed on to \code{\link[edgeR]{DGEList}},
#'  and you may find further details or examples in its documentation.
#'  
#' @param method (Optional). The label of the edgeR-implemented normalization to use.
#'  See \code{\link[edgeR]{calcNormFactors}} for supported options and details. 
#'  The default option is \code{'RLE'}, which is a scaling factor method 
#'  proposed by Anders and Huber (2010).
#'  At time of writing, the \link[edgeR]{edgeR} package supported 
#'  the following options to the \code{method} argument:
#'  
#'  \code{c('TMM', 'RLE', 'upperquartile', 'none')}.
#'
#' @param ... Additional arguments passed on to \code{\link[edgeR]{DGEList}}
#' 
#' @examples
#' 
phyloseq_to_edgeR = function(physeq, group, method = "RLE", ...) {
  require("edgeR")
  require("phyloseq")
  # Enforce orientation.
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)
  }
  x = as(otu_table(physeq), "matrix")
  # Add one to protect against overflow, log(0) issues.
  x = x + 1
  # Check `group` argument
  if (identical(all.equal(length(group), 1), TRUE) & nsamples(physeq) > 1) {
    # Assume that group was a sample variable name (must be categorical)
    group = get_variable(physeq, group)
  }
  # Define gene annotations (`genes`) as tax_table
  taxonomy = tax_table(physeq, errorIfNULL=FALSE)
  if( !is.null(taxonomy) ){
    taxonomy = data.frame(as(taxonomy, "matrix"))
  } 
  # Now turn into a DGEList
  y = DGEList(counts = x, group = group, genes = taxonomy, remove.zeros = TRUE, 
              ...)
  # Calculate the normalization factors
  z = calcNormFactors(y, method = method)
  # Check for division by zero inside `calcNormFactors`
  if (!all(is.finite(z$samples$norm.factors))) {
    stop("Something wrong with edgeR::calcNormFactors on this data,\n         non-finite $norm.factors, consider changing `method` argument")
  }
  # Estimate dispersions
  return(estimateTagwiseDisp(estimateCommonDisp(z)))
}


```
#No filering profile community
##Domain
```{r Set up domain}
my_domain_phyloseq = Fiord_phyloseq %>% #Identify the original subsetted phyloseq object
  tax_glom("Domain") %>% #Merge species with the same Domain.
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Convert the absolute count to a percentage count
  psmelt() %>% #Adjust the data from a phyloseq object into a data.frame-class table  
  filter(Abundance > 0.01)   %>% #Remove samples that make up less than one percent of the abundance.
  arrange(Domain)#Arange the samples by Domain.
```
```{r Plotting Domain}
Plot_location_Domain = ggplot(my_domain_phyloseq,
                          aes(x=as.factor(Depth_m),
                              y=Abundance,
                              fill=Domain)) +
  geom_bar(stat = "identity", 
           position = "fill") +
  scale_y_continuous(labels = percent_format()) +
  #scale_fill_manual("Genus", values = Relativeabundancecolourlist)+
  #ggtitle("Fjord Sound sample") + 
  #facet_grid("Sample_Depth")+
  #theme(plot.title = element_text(hjust = 0.5))+
  xlab("Sample Depth (m)")+
  ylab("Relative abundance (%)")

```
```{r Set up phyla level}
#Subset to only eukaryota
Subseteukaryota = c(as.character("Eukaryota"))

Eukaryota_phyloseq = subset_taxa(Fiord_phyloseq, Domain == "Eukaryota")
#Make a phyloseq based on purely domain from eukaryotes
my_phylum_phyloseq = Eukaryota_phyloseq %>% #Identify the original subsetted phyloseq object
  tax_glom("Phylum") %>% #Merge species with the same Genus.
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Convert the absolute count to a percentage count
  psmelt() %>% #Adjust the data from a phyloseq object into a data.frame-class table  
  filter(Abundance > 0.01)   %>% #Remove samples that make up less than one percent of the abundance.
  arrange(Phylum)#Arange the samples by genus.

```
```{r Plot phyla level}
Plot_location_Phyla = ggplot(my_phylum_phyloseq,
                          aes(x=as.factor(Depth_m),
                              y=Abundance,
                              fill=Phylum)) +
  geom_bar(stat = "identity", 
           position = "fill") +
  scale_y_continuous(labels = percent_format()) +
  xlab("Sample Depth (m)")+
  ylab("Relative abundance (%)")

```
```{r Export rough plots}
saveRDS(Plot_location_Domain, "~/Desktop/Domainprofile18S.rds")
saveRDS(Plot_location_Phyla, "~/Desktop/Phylaprofile18S.rds")

```
#Signficantly changing horizontal community
##Correlated with location
```{r Set up horizontal location correlation}
#Make phyloseq of Transect
Fiord_phyloseq_Transect_Euk = subset_samples(Eukaryota_phyloseq, Depth_m == "0" | Depth_m == "1" |Depth_m == "10")

Fiord_phyloseq_Transect_Euk@sam_data$Depth_m[Fiord_phyloseq_Transect_Euk@sam_data$Depth_m == 1] = 0

#Group them by their station
dge_EdgeR_obj_Transect_Euk = phyloseq_to_edgeR(Fiord_phyloseq_Transect_Euk, group = "Distance")

# Perform binary test
et_EdgeR_Transect_Euk = exactTest(dge_EdgeR_obj_Transect_Euk)
# Extract values from test results
tt_EdgeR_Transect_Euk = topTags(et_EdgeR_Transect_Euk, n = nrow(dge_EdgeR_obj_Transect_Euk$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Transect_Euk = tt_EdgeR_Transect_Euk@.Data[[1]]
sigtab_2fold_EdgeR_Transect_Euk<- subset(res_EdgeR_Transect_Euk, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1
sigtab_2fold_FDR_Transect_Euk <- subset(sigtab_2fold_EdgeR_Transect_Euk, FDR < 0.1)


keepTaxa_FDR_Transect_Euk <- sigtab_2fold_EdgeR_Transect_Euk$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Transect_Euk <- subset_taxa(Fiord_phyloseq_Transect_Euk, Genus %in% keepTaxa_FDR_Transect_Euk) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Transect_Phylum_Euk <- tax_glom(Twofold_FDR_Transect_Euk, taxrank = 'Phylum') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Transect_Phylum_Euk <- dat_2fold_FDR_Transect_Phylum_Euk[order(dat_2fold_FDR_Transect_Phylum_Euk$Phylum),] #Order them at the Phylum level

dat_2fold_FDR_Transect_Phylum_Euk$Phylum <- as.character(dat_2fold_FDR_Transect_Phylum_Euk$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Transect_Phylum_Euk <- ddply(dat_2fold_FDR_Transect_Phylum_Euk, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phylum whose rel. abund. is less than 1%
remainder_Transect_Phylum_Euk <- medians_Transect_Phylum_Euk[medians_Transect_Phylum_Euk$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Transect_Phylum_Euk[dat_2fold_FDR_Transect_Phylum_Euk$Phylum %in% remainder_Transect_Phylum_Euk,]$Phylum <- 'RareTaxa'

Summary_Transect_Phylum_Euk <- summarySE(dat_2fold_FDR_Transect_Phylum_Euk, measurevar="Abundance", groupvars=c("Phylum", "Depth_m", "Distance"))
Summary_Transect_Phylum_Euk

 

Summary_Transect_Phylum_Euk<-dplyr::arrange(Summary_Transect_Phylum_Euk,Phylum, Abundance,Distance)

Summary_Transect_Phylum_Euk$Phylum <- factor(Summary_Transect_Phylum_Euk$Phylum,
                         levels=(unique(Summary_Transect_Phylum_Euk$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Transect_Phylum_Euk$Phylum)

#Rename levels
levels(Summary_Transect_Phylum_Euk$Phylum) = c("Opisthokonta",
                                  "Rare Taxa (<1%)",
                                  "SAR")

#Check to make sure it worked
levels(Summary_Transect_Phylum_Euk$Phylum) 

#Save as table for ease of reading.
write.csv(Summary_Transect_Phylum_Euk, file = "Ordered_summary_Transect_Phylumlvl_Euk.csv")

```
```{r Plot significantly correlated horizontal location community}

Summary_plot_Transect_Phylum_Euk<-ggplot(Summary_Transect_Phylum_Euk, 
                        aes(x=as.numeric(as.character(Distance)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Sample Site")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)+
  facet_wrap(~Depth_m)


Summary_plot_Transect_Phylum_Euk

```
```{r Significant vs. non-significant community - Horizontal location}
# get abundance in %
Sample_counts <- transform_sample_counts(Fiord_phyloseq_Transect_Euk, function(x) x/sum(x))

# agglomerate taxa
glom <- tax_glom(Sample_counts, taxrank = 'Genus')

# create dataframe from phyloseq object
dat <- psmelt(glom)


# convert Phylum to a character vector from a factor because R
dat$Genus <- as.character(dat$Genus)

#Convert to character, because R
Significant_taxa <- c(as.character(keepTaxa_FDR_Transect_Euk))  #Extract the OTU table that was shown to be significant

#Change the name of the significant organisms
dat[dat$Genus %in% Significant_taxa,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df = subset(dat, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat[dat$Genus %!in% Significant_taxa,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df = subset(dat, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign = rbind(Nonsignificant_df, Significant_df)

#Plot
plotme = ggplot(plotsign.vs.nonSign,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme

dev.copy(png, "Sign.vs.NonSign", units = "in", width = 5, height = 4, res = 500)
dev.off()

```
##Correlated with salinity
```{r Set up horizontal salinity  correlation}

Summary_Transect_Phylum_Sal = read.csv("Ordered_summary_Transect_Phylumlvl_Sal_18S.csv")

```
```{r Plot significantly correlated horizontal salinity community}
Summary_plot_Transect_Phylum_Sal = ggplot(Summary_Transect_Phylum_Sal, 
                        aes(x=as.numeric(as.character(Distance)), 
                            y=Abundance*100, 
                            colour=Phylum, 
                            group = Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank(),
        axis.title.y = element_blank())+
  xlab("Km from the outermost sample")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)+
  facet_wrap(~Depth_m)
```
```{r Significant vs. non-significant community - Horizontal salinity}
# get abundance in %
Sample_counts_Transect <- transform_sample_counts(Fiord_phyloseq_Transect, function(x) x/sum(x))

# agglomerate taxa
glom_Transect <- tax_glom(Sample_counts_Transect, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Transect <- psmelt(glom_Transect)


# convert Phylum to a character vector from a factor because R
dat_Transect$Genus <- as.character(dat_Transect$Genus)

#Convert to character, as R
Significant_taxa_Transect_Sal <- as.character(keepTaxa_FDR_Transect_Sal)

#Change the name of the significant organisms
dat_Transect[dat_Transect$Genus %in% Significant_taxa_Transect_Sal,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Transect_Sal = subset(dat_Transect, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Transect[dat_Transect$Genus %!in% Significant_df_Transect_Sal,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Transect_Sal = subset(dat_Transect, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Transect_Sal = rbind(Nonsignificant_df_Transect_Sal, Significant_df_Transect_Sal)

#Plot
plotme_Transect_Sal = ggplot(plotsign.vs.nonSign_Transect_Sal,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Transect_Sal

dev.copy(png, "Sign.vs.NonSign_Transect_Sal", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Transect_Sal$Genus))

#Total number of sample
 10635       +     2970
#13605

#Ratio of significant
2970/13605
#21.8

#Ratio of non-significant
10635/13605
#78.2

```
##Export for ggarrange
```{r Export horizontal}
saveRDS(Summary_plot_Transect_Phylum_Euk,"~/Desktop/Phylum_EdgeR_Transect_18S.rds")
saveRDS(Summary_plot_Transect_Phylum_Sal,"~/Desktop/Phylum_EdgeR_Profile_Sal_18S.rds")


```
#Signficantly changing vertical community
##Correlated with depth
```{r Set up vertical depth correlation}
#Isolate eukaryotes
Eukaryota_phyloseq = subset_taxa(Fiord_phyloseq, Domain == "Eukaryota")

#Make phyloseq of profile
Fiord_phyloseq_profile_Euk = subset_samples(Eukaryota_phyloseq, Layer == "5")

#Group them by their depths
dge_EdgeR_obj_Profile_Euk = phyloseq_to_edgeR(Fiord_phyloseq_profile_Euk, group = "Depth_m")

# Perform binary test
et_EdgeR_Profile_Euk = exactTest(dge_EdgeR_obj_Profile_Euk)
# Extract values from test results
tt_EdgeR_Profile_Euk = topTags(et_EdgeR_Profile_Euk, n = nrow(dge_EdgeR_obj_Profile_Euk$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Profile_Euk = tt_EdgeR_Profile_Euk@.Data[[1]]
sigtab_2fold_EdgeR_Profile_Euk<- subset(res_EdgeR_Profile_Euk, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1
sigtab_2fold_FDR_Profile_Euk  <- subset(sigtab_2fold_EdgeR_Profile_Euk , FDR < 0.1)


keepTaxa_FDR_Profile_Euk  <- sigtab_2fold_EdgeR_Profile_Euk$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Profile_Euk  <- subset_taxa(Fiord_phyloseq_profile_Euk , Genus %in% keepTaxa_FDR_Profile_Euk) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Profile_Phylum_Euk  <- tax_glom(Twofold_FDR_Profile_Euk, taxrank = 'Phylum') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Profile_Phylum_Euk  <- dat_2fold_FDR_Profile_Phylum_Euk [order(dat_2fold_FDR_Profile_Phylum_Euk$Phylum),] #Order them at the Phylum level

dat_2fold_FDR_Profile_Phylum_Euk$Phylum <- as.character(dat_2fold_FDR_Profile_Phylum_Euk$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Profile_Phylum_Euk <- ddply(dat_2fold_FDR_Profile_Phylum_Euk, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phylum whose rel. abund. is less than 1%
remainder_Profile_Phylum_Euk <- medians_Profile_Phylum_Euk[medians_Profile_Phylum_Euk$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Profile_Phylum_Euk[dat_2fold_FDR_Profile_Phylum_Euk$Phylum %in% remainder_Profile_Phylum_Euk,]$Phylum <- 'RareTaxa'

Summary_Profile_Phylum_Euk <- summarySE(dat_2fold_FDR_Profile_Phylum_Euk, measurevar="Abundance", groupvars=c("Phylum", "Depth_m"))
Summary_Profile_Phylum_Euk

 

Summary_Profile_Phylum_Euk<-dplyr::arrange(Summary_Profile_Phylum_Euk,Phylum, Abundance)

Summary_Profile_Phylum_Euk$Phylum <- factor(Summary_Profile_Phylum_Euk$Phylum,
                         levels=(unique(Summary_Profile_Phylum_Euk$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Profile_Phylum_Euk$Phylum)

#Rename levels
levels(Summary_Profile_Phylum_Euk$Phylum) = c("Opisthokonta",
                                              "Rare Taxa (<1%)",
                                              "SAR")

#Check to make sure it worked
levels(Summary_Profile_Phylum_Euk$Phylum) 

#Save as table for ease of reading.
write.csv(Summary_Profile_Phylum_Euk, file = "Ordered_summary_Profile_Phylumlvl_Euk.csv")

```
```{r Plot significantly correlated vertical depth community}

Summary_plot_Profile_Phylum_Euk <-ggplot(Summary_Profile_Phylum_Euk, 
                        aes(x=as.numeric(as.character(Depth_m)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black"#, 
                #position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Sample Depth (m)")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)

```
```{r Significant vs. non-significant community - vertical depth}
# get abundance in %
Sample_counts_Profile <- transform_sample_counts(Fiord_phyloseq_profile, function(x) x/sum(x))

# agglomerate taxa
glom_Profile <- tax_glom(Sample_counts_Profile, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Profile <- psmelt(glom_Profile)


# convert Phylum to a character vector from a factor because R
dat_Profile$Genus <- as.character(dat_Profile$Genus)

#Convert to character, as R
Significant_taxa_Profile <- as.character(keepTaxa_FDR_Profile_Euk)

#Change the name of the significant organisms
dat_Profile[dat_Profile$Genus %in% Significant_taxa_Profile,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Profile = subset(dat_Profile, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Profile[dat_Profile$Genus %!in% Significant_df_Profile,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Profile = subset(dat_Profile, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Profile = rbind(Nonsignificant_df_Profile, Significant_df_Profile)

#Plot
plotme_Profile = ggplot(plotsign.vs.nonSign_Profile,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Profile

dev.copy(png, "Sign.vs.NonSign_Profile", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Profile$Genus))

```
##Correlated with salinity
```{r Set up vertical salinity  correlation}
Summary_Profile_Phylum_Sal = read.csv("Ordered_summary_Profile_Phylalvl_Sal_18S.csv")
```
```{r Plot significantly correlated vertical salinity community}
Summary_plot_Profile_Phylum_Sal <-ggplot(Summary_Profile_Phylum_Sal, 
                        aes(x=as.factor(Depth_m), 
                            y=Abundance*100, 
                            colour=Phylum,
                            group = Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank(),
        axis.title.y = element_blank())+
  xlab("Sample Depth (m)")+
  ylab("Mean relative abundance (%)")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)

```
```{r Significant vs. non-significant community - vertical salinity}
# get abundance in %
Sample_counts_Profile <- transform_sample_counts(Fiord_phyloseq_profile, function(x) x/sum(x))

# agglomerate taxa
glom_Profile <- tax_glom(Sample_counts_Profile, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Profile <- psmelt(glom_Profile)


# convert Phylum to a character vector from a factor because R
dat_Profile$Genus <- as.character(dat_Profile$Genus)

#Convert to character, as R
Significant_taxa_Profile <- as.character(keepTaxa_FDR_Profile_Sal)

#Change the name of the significant organisms
dat_Profile[dat_Profile$Genus %in% Significant_taxa_Profile,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Profile = subset(dat_Profile, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Profile[dat_Profile$Genus %!in% Significant_df_Profile,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Profile = subset(dat_Profile, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Profile = rbind(Nonsignificant_df_Profile, Significant_df_Profile)

#Plot
plotme_Profile = ggplot(plotsign.vs.nonSign_Profile,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Profile

dev.copy(png, "Sign.vs.NonSign_Profile_Sal", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Profile$Genus))

#Total number of sample
 8508       +     2868
#11376

#Ratio of significant
2868/11376
#25.2

#Ratio of non-significant
8508/11376
#74.8
```

##Export for ggarrange
```{r Export vertical}
saveRDS(Summary_plot_Profile_Phylum_Euk,"~/Desktop/Phylum_EdgeR_Profile_18S.rds")
saveRDS(Summary_plot_Profile_Phylum_Sal_Mod,"~/Desktop/Phylum_EdgeR_Profile_Sal_18S.rds")
```




