---
title: "16S significantly changing community"
author: "Sven"
date: "2/25/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
##Notes:
It is best to go through each dataset (consensus fjord/horizontal Long Sound community/ Vertical Long Sound community) one by one. So following the order of this file from start to finish, in order.

#Set up
```{r Load packages}
library(vegan)
library(phyloseq)
library(ggplot2)
library(dplyr)
library(Rmisc)
library(plyr)
```
##Import files:
```{r Set path}

uzdir <- "/Users/sven/Desktop/Project_R_analysis/"
otutable_biom_file <- paste("/Users/sven/Desktop/Project_R_analysis/", "Merged_otu_table_json.biom", sep = "")
map_file <- paste("/Users/sven/Desktop/Project_R_analysis/Sorted_out/", "Modified_mapping_file_mod_with_names_and_variables.txt", sep = "")


```
```{r Create phyloseq object}
# Now import the .biom-formatted otu_table-tax_table file.

biom_otu_tax <- import_biom(otutable_biom_file)

# Add sample data to the dataset using merge
bmsd <- import_qiime_sample_data(map_file)
class(bmsd)
dim(bmsd)
sample_data(bmsd)
sample_variables(bmsd)
biom_otu_tax


#Merge into phyloseq format
Fiord_phyloseq <- merge_phyloseq(biom_otu_tax, bmsd)
Fiord_phyloseq
sample_sums(Fiord_phyloseq)

#Save original phyloseq file
Fiord_phyloseq_v = Fiord_phyloseq

#Fix Misslabel
Fiord_phyloseq@sam_data$Sample_Depth[Fiord_phyloseq@sam_data$Sample_Depth == 1] = 0
#Ensure sample depth is seen as a factor for ease of use
sample_data(Fiord_phyloseq)$Sample_Depth<-as.factor(sample_data(Fiord_phyloseq)$Sample_Depth)
```
##Create average result for multiple rarefaction by transforming data using (divide by 10) and check counts per sample
```{r Create average result for multiple rarefaction by transforming data using (divide by 10), results='markup'}
Fiord_phyloseq = transform_sample_counts(Fiord_phyloseq, function(x) x/10)
sample_sums(Fiord_phyloseq)
```
#### Round and confirm count number
```{r Round and confirm count number, results='markup'}
Fiord_phyloseq = transform_sample_counts(Fiord_phyloseq, round)
sample_sums(Fiord_phyloseq)
Fiord_phyloseq = prune_samples(sample_sums(Fiord_phyloseq)>=1, Fiord_phyloseq)
sample_sums(Fiord_phyloseq)
```
Check that all OTUs have representative counts  
For here taxa = OTU  
__Commands interpretation:__  
_Total number of taxa in dataset:_ sum(taxa_sums(Fiord_phyloseq) > 0)   

_Any taxa with no hits:_ any(taxa_sums(Fiord_phyloseq)== 0)
```{r identify taxa with only zeros, results='markup', echo=TRUE}
sum(taxa_sums(Fiord_phyloseq) > 0)
any(taxa_sums(Fiord_phyloseq)== 0)
sum(taxa_sums(Fiord_phyloseq) == 0)
any(taxa_sums(Fiord_phyloseq) > 1)
sum(taxa_sums(Fiord_phyloseq) > 1)
any(taxa_sums(Fiord_phyloseq) < 1)
sum(taxa_sums(Fiord_phyloseq) < 1)
```
####Prune taxa with less than 1 count and check taxa numbers again
```{r  Save original file and create new file with only present (no zeroes) taxa, results='markup', echo=TRUE}

#Create new file with only present (no zeroes) taxa

Fiord_phyloseq = prune_taxa(taxa_sums(Fiord_phyloseq) > 1, Fiord_phyloseq)
any(sample_sums(Fiord_phyloseq) == 0)
any(sample_sums(Fiord_phyloseq) > 0)
sum(taxa_sums(Fiord_phyloseq) > 0)
any(sample_sums(Fiord_phyloseq) < 1)
sum(taxa_sums(Fiord_phyloseq) < 1)
```
##Compare sequences per sample or OTU
```{r Compare sequences per sample or OTU}
readsumsdf = data.frame(nreads = sort(taxa_sums(Fiord_phyloseq),TRUE), sorted = 1:ntaxa(Fiord_phyloseq), type = "OTU")
readsumsdf = rbind(readsumsdf,data.frame(nreads = sort(sample_sums(Fiord_phyloseq),TRUE),sorted = 1:nsamples(Fiord_phyloseq), type = "Samples"))

title = "Total number of reads"

p = ggplot(readsumsdf, aes(x = sorted, y = nreads)) +
  geom_bar(stat = "identity")

p + ggtitle(title) +
  scale_y_log10() +
  facet_wrap(~type, 1, scales = "free")
```
```{r What sample variables exist in the phyloseq data}
sample_variables(Fiord_phyloseq)
```
```{r Attached OTU ID}
tax_table(Fiord_phyloseq) <- cbind(tax_table(Fiord_phyloseq), OTU=taxa_names(Fiord_phyloseq))
```
```{r Rename Ranks}
colnames(tax_table(Fiord_phyloseq)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species", "OTU")
tax_table(Fiord_phyloseq) =gsub("D_0__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_1__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_2__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_3__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_4__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_5__", "", tax_table(Fiord_phyloseq))
tax_table(Fiord_phyloseq) =gsub("D_6__", "", tax_table(Fiord_phyloseq))

```
#Subset
```{r Subset}
#Subset to in/out
Fiord_phyloseq_in_out = subset_samples(Fiord_phyloseq, !Sample_type == "Transverse")

#Subset horizontal community
Fiord_phyloseq_Horonly = subset_samples(Fiord_phyloseq, Sample_type == "Transverse" & !Sample_Depth == "40" & !Sample_Depth == "100" & !Sample_Depth == "200" & !Sample_Depth == "360")

#Subset vertical community
Fiord_phyloseq_Vert = subset_samples(Fiord_phyloseq, !Sample_type == "in" & !Sample_type == "out")
Fiord_phyloseq_Vert = subset_samples(Fiord_phyloseq_Vert, Sample_Site == "L5")
```
##Phylum colour list for consistency
```{r Phylum colour list}
Phylum_colour_list = c(Acidobacteria = "red",
                       Actinobacteria = "green",
                       "Alphaproteobacteria" = "burlywood1",
                       "Alveolata" = "blueviolet",
                       Archaeplastida = "#F8766D",
                       Bacteroidetes = "steelblue",  
                       Chlamydiae = "navy", 
                       Chloroflexi = "grey",
                       Cyanobacteria = "purple", 
                       "Deltaproteobacteria" = "orange",
                       "Discoba" = "darkslategray1",
                       Euryarchaeota = "aquamarine", 
                       "Epsilonproteobacteria" = "darkslateblue",
                       "Excavata" = "#7CAE00",
                       "Gammaproteobacteria" = "dimgray",
                       "Holozoa" = "forestgreen",
                       "Marine Group I" = "gainsboro",
                       Marinimicrobia = "darkseagreen1",
                       Opisthokonta = "magenta", 
                       Parcubacteria = "chocolate2", 
                       Planctomycetes = "wheat", 
                       Proteobacteria = "olivedrab",
                       "Rhizaria" = "goldenrod",
                       Thaumarchaeota = "firebrick", 
                       SAR = "darkred", 
                       "Stramenopiles" = "darkorchid1",
                       "Thermoplasmata" = "tan",
                       Verrucomicrobia = "gold4", 
                       "Rare Taxa (<1%)" = "black")
```
#Exact test function
```{r Exact test code}

#check open packages
(.packages())
##Close all but phyloseq
detachAllPackages <- function() {
  
  basic.packages <- c("package:stats","package:graphics","package:grDevices","package:utils","package:datasets","package:methods","package:base")
  
  package.list <- search()[ifelse(unlist(gregexpr("package:",search()))==1,TRUE,FALSE)]
  
  package.list <- setdiff(package.list,basic.packages)
  
  if (length(package.list)>0)  for (package in package.list) detach(package, character.only=TRUE)
  
}

detachAllPackages()


library("phyloseq")
packageVersion("phyloseq")
library("edgeR")
packageVersion("edgeR")
library(phyloseq)
library(ggplot2)
library(plyr)
library(scales)
library(reshape)
library(RColorBrewer)
library(grid)
library(empiricalFDR.DESeq2)
library("DESeq2")
library(dplyr)
library(Rmisc)


#' Convert phyloseq OTU count data into DGEList for edgeR package
#' 
#' Further details.
#' 
#' @param physeq (Required).  A \code{\link{phyloseq-class}} or
#'  an \code{\link{otu_table-class}} object. 
#'  The latter is only appropriate if \code{group} argument is also a 
#'  vector or factor with length equal to \code{nsamples(physeq)}.
#'  
#' @param group (Required). A character vector or factor giving the experimental
#'  group/condition for each sample/library. Alternatively, you may provide
#'  the name of a sample variable. This name should be among the output of
#'  \code{sample_variables(physeq)}, in which case
#'  \code{get_variable(physeq, group)} would return either a character vector or factor.
#'  This is passed on to \code{\link[edgeR]{DGEList}},
#'  and you may find further details or examples in its documentation.
#'  
#' @param method (Optional). The label of the edgeR-implemented normalization to use.
#'  See \code{\link[edgeR]{calcNormFactors}} for supported options and details. 
#'  The default option is \code{'RLE'}, which is a scaling factor method 
#'  proposed by Anders and Huber (2010).
#'  At time of writing, the \link[edgeR]{edgeR} package supported 
#'  the following options to the \code{method} argument:
#'  
#'  \code{c('TMM', 'RLE', 'upperquartile', 'none')}.
#'
#' @param ... Additional arguments passed on to \code{\link[edgeR]{DGEList}}
#' 
#' @examples
#' 
phyloseq_to_edgeR = function(physeq, group, method = "RLE", ...) {
  require("edgeR")
  require("phyloseq")
  # Enforce orientation.
  if (!taxa_are_rows(physeq)) {
    physeq <- t(physeq)
  }
  x = as(otu_table(physeq), "matrix")
  # Add one to protect against overflow, log(0) issues.
  x = x + 1
  # Check `group` argument
  if (identical(all.equal(length(group), 1), TRUE) & nsamples(physeq) > 1) {
    # Assume that group was a sample variable name (must be categorical)
    group = get_variable(physeq, group)
  }
  # Define gene annotations (`genes`) as tax_table
  taxonomy = tax_table(physeq, errorIfNULL=FALSE)
  if( !is.null(taxonomy) ){
    taxonomy = data.frame(as(taxonomy, "matrix"))
  } 
  # Now turn into a DGEList
  y = DGEList(counts = x, group = group, genes = taxonomy, remove.zeros = TRUE, 
              ...)
  # Calculate the normalization factors
  z = calcNormFactors(y, method = method)
  # Check for division by zero inside `calcNormFactors`
  if (!all(is.finite(z$samples$norm.factors))) {
    stop("Something wrong with edgeR::calcNormFactors on this data,\n         non-finite $norm.factors, consider changing `method` argument")
  }
  # Estimate dispersions
  return(estimateTagwiseDisp(estimateCommonDisp(z)))
}


```
#Signficantly changing horizontal community
##Correlated with location
```{r Set up horizontal location correlation}

#Add distance to phyloseq object
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L1", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = 0
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L2", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(5.59)
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L3", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(14.3)
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L4", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(10.67)
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L5", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(8.47)
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L6", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(4.73)
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L7", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(3.16)
Fiord_phyloseq_Horonly@sam_data[["Distance_from_L1"]][grep("L8", Fiord_phyloseq_Horonly@sam_data[["Sample_Site"]])] = as.numeric(2.47)

#Group them by their Distance
dge_EdgeR_obj_Transect_Dist = phyloseq_to_edgeR(Fiord_phyloseq_Horonly, group = "Distance_from_L1")

# Perform binary test
et_EdgeR_Transect_Dist = exactTest(dge_EdgeR_obj_Transect_Dist)
# Extract values from test results
tt_EdgeR_Transect_Dist = topTags(et_EdgeR_Transect_Dist, n = nrow(dge_EdgeR_obj_Transect_Dist$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Transect_Dist = tt_EdgeR_Transect_Dist@.Data[[1]]
sigtab_2fold_EdgeR_Transect_Dist<- subset(res_EdgeR_Transect_Dist, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)


#Keep only FDR corrected <.1
sigtab_2fold_FDR_Transect_Dist <- subset(sigtab_2fold_EdgeR_Transect_Dist, FDR < 0.1)

keepTaxa_FDR_Transect_Dist <- sigtab_2fold_EdgeR_Transect_Dist$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Transect_Dist <- subset_taxa(Fiord_phyloseq_Horonly, Genus %in% keepTaxa_FDR_Transect_Dist) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Transect_Dist <- tax_glom(Twofold_FDR_Transect_Dist, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Transect_Dist <- dat_2fold_FDR_Transect_Dist[order(dat_2fold_FDR_Transect_Dist$Family),] #Order them at the Phylum level

dat_2fold_FDR_Transect_Dist$Phylum <- as.character(dat_2fold_FDR_Transect_Dist$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Transect_Dist <- ddply(dat_2fold_FDR_Transect_Dist, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Transect_Dist <- medians_Transect_Dist[medians_Transect_Dist$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Transect_Dist[dat_2fold_FDR_Transect_Dist$Phylum %in% remainder_Transect_Dist,]$Phylum <- 'RareTaxa'


Summary_Transect_Dist <- summarySE(dat_2fold_FDR_Transect_Dist, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "Distance_from_L1"))
Summary_Transect_Dist

write.csv(Summary_Transect_Dist, file = "Summary_for_Nadjejda_Transect_16S_Dist.csv")

Summary_Transect_Dist<-dplyr::arrange(Summary_Transect_Dist,Phylum, Abundance)

Summary_Transect_Dist$Phylum <- factor(Summary_Transect_Dist$Phylum,
                         levels=(unique(Summary_Transect_Dist$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Transect_Dist$Phylum)

#Rename levels for proper rare taxa label
levels(Summary_Transect_Dist$Phylum) = c("Actinobacteria",
                                         "Bacteroidetes",
                                         "Cyanobacteria",
                                         "Planctomycetes",
                                         "Proteobacteria",
                                         "Rare Taxa (<1%)",
                                         "Thaumarchaeota",
                                         "Verrucomicrobia")

#Check to make sure it worked
levels(Summary_Transect_Dist$Phylum) 

#Save as table for ease of reading.
write.csv(Summary_Transect_Dist, file = "Ordered_summary_Transect_phylalvl_Dist.csv")

#Set up position dodge
pd = position_dodge(0.15)

```
```{r Plot significantly correlated horizontal location community}

Summary_plot_Transect_Dist<-ggplot(Summary_Transect_Dist, 
                        aes(x=as.numeric(as.character(Distance_from_L1)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Km from the outermost sample")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)+
  facet_wrap(~Sample_Depth)


Summary_plot_Transect_Dist

```
```{r Significant vs. non-significant community - Horizontal location}

# get abundance in %
Sample_counts_Transect_Dist <- transform_sample_counts(Fiord_phyloseq_Horonly, function(x) x/sum(x))

# agglomerate taxa
glom_Transect_Dist <- tax_glom(Sample_counts_Transect_Dist, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Transect_Dist <- psmelt(glom_Transect_Dist)


# convert Phylum to a character vector from a factor because R
dat_Transect_Dist$Genus <- as.character(dat_Transect_Dist$Genus)

#Convert to character, as R
Significant_taxa_Transect_Dist <- as.character(keepTaxa_FDR_Transect_Dist)

#Change the name of the significant organisms
dat_Transect_Dist[dat_Transect_Dist$Genus %in% Significant_taxa_Transect_Dist,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Transect_Dist = subset(dat_Transect_Dist, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Transect_Dist[dat_Transect_Dist$Genus %!in% Significant_df_Transect_Dist,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Transect_Dist = subset(dat_Transect_Dist, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Transect_Dist = rbind(Nonsignificant_df_Transect_Dist, Significant_df_Transect_Dist)

#Plot
plotme_Transect_Dist = ggplot(plotsign.vs.nonSign_Transect_Dist,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Transect_Dist

dev.copy(png, "Sign.vs.NonSign_Transect_Dist", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Transect_Dist$Genus))

#Total number of sample
 40043       +     16238
#56281

#Ratio of significant
16238/56281
#28.9

#Ratio of non-significant
40043/56281
#71.1

```
##Correlated with salinity
```{r Set up horizontal salinity  correlation}

#Add distance to phyloseq object
##Read in .csv file
CTDtoadd = read.csv("/Users/sven/Desktop/Hor_CTDtoAdd.csv", row.names = 1)

Arranged_CTD = plyr::arrange(CTDtoadd, SampleID)
Arranged_CTD$Salinity
Arranged_CTD$Temperature
Arranged_CTD$OxygenPerSat

##Merge CTD df and sample data
CTD_df = merge(CTDtoadd, Fiord_phyloseq_Horonly@sam_data)
row.names(CTD_df) = CTD_df$SampleID

#Fiord_phyloseq_Horonly = Fiord_phyloseq_Horonly_v0 #Insurace in case of muck ups
##Replace the old sample data with the new df
Fiord_phyloseq_Horonly@sam_data = sample_data(CTD_df)

#Group them by their Salinity
dge_EdgeR_obj_Transect_Sal = phyloseq_to_edgeR(Fiord_phyloseq_Horonly, group = "Salinity")

# Perform binary test
et_EdgeR_Transect_Sal = exactTest(dge_EdgeR_obj_Transect_Sal)
# Extract values from test results
tt_EdgeR_Transect_Sal = topTags(et_EdgeR_Transect_Sal, n = nrow(dge_EdgeR_obj_Transect_Sal$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Transect_Sal = tt_EdgeR_Transect_Sal@.Data[[1]]
sigtab_2fold_EdgeR_Transect_Sal<- subset(res_EdgeR_Transect_Sal, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)


#Keep only FDR corrected <.1
sigtab_2fold_FDR_Transect_Sal <- subset(sigtab_2fold_EdgeR_Transect_Sal, FDR < 0.1)


keepTaxa_FDR_Transect_Sal <- sigtab_2fold_EdgeR_Transect_Sal$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Transect_Sal <- subset_taxa(Fiord_phyloseq_Horonly, Genus %in% keepTaxa_FDR_Transect_Sal) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Transect_Sal <- tax_glom(Twofold_FDR_Transect_Sal, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Transect_Sal <- dat_2fold_FDR_Transect_Sal[order(dat_2fold_FDR_Transect_Sal$Family),] #Order them at the Phylum level

dat_2fold_FDR_Transect_Sal$Phylum <- as.character(dat_2fold_FDR_Transect_Sal$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Transect_Sal <- ddply(dat_2fold_FDR_Transect_Sal, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Transect_Sal <- medians_Transect_Sal[medians_Transect_Sal$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Transect_Sal[dat_2fold_FDR_Transect_Sal$Phylum %in% remainder_Transect_Sal,]$Phylum <- 'RareTaxa'

Summary_Transect_Sal <- summarySE(dat_2fold_FDR_Transect_Sal, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "Salinity", "Distance_from_L1"))
Summary_Transect_Sal

write.csv(Summary_Transect_Sal, file = "Summary_for_Nadjejda_Transect_16S_Sal.csv")

Summary_Transect_Sal<-dplyr::arrange(Summary_Transect_Sal,Phylum, Abundance)

Summary_Transect_Sal$Phylum <- factor(Summary_Transect_Sal$Phylum,
                         levels=(unique(Summary_Transect_Sal$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Transect_Sal$Phylum)

#Rename levels
levels(Summary_Transect_Sal$Phylum) = c("Actinobacteria",
                                         "Bacteroidetes",
                                         "Cyanobacteria",
                                         "Marinimicrobia",
                                         "Planctomycetes",
                                         "Proteobacteria",
                                         "Rare Taxa (<1%)",
                                         "Thaumarchaeota",
                                         "Verrucomicrobia")

#Check to make sure it worked
levels(Summary_Transect_Sal$Phylum) 

#Save as table for ease of reading.
write.csv(Summary_Transect_Sal, file = "Ordered_summary_Transect_phylalvl_Sal.csv")

```
```{r Plot significantly correlated horizontal salinity community}
Summary_plot_Transect_Sal<-ggplot(Summary_Transect_Sal, 
                        aes(x=as.numeric(as.character(Distance_from_L1)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Km from the outermost sample")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)+
  facet_wrap(~Sample_Depth)


Summary_plot_Transect_Sal
```
```{r Significant vs. non-significant community - Horizontal salinity}
# get abundance in %
Sample_counts_Transect_Sal <- transform_sample_counts(Fiord_phyloseq_Horonly, function(x) x/sum(x))

# agglomerate taxa
glom_Transect_Sal <- tax_glom(Sample_counts_Transect_Sal, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Transect_Sal <- psmelt(glom_Transect_Sal)


# convert Phylum to a character vector from a factor because R
dat_Transect_Sal$Genus <- as.character(dat_Transect_Sal$Genus)

#Convert to character, as R
Significant_taxa_Transect_Sal <- as.character(keepTaxa_FDR_Transect_Sal)

#Change the name of the significant organisms
dat_Transect_Sal[dat_Transect_Sal$Genus %in% Significant_taxa_Transect_Sal,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Transect_Sal = subset(dat_Transect_Sal, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Transect_Sal[dat_Transect_Sal$Genus %!in% Significant_df_Transect_Sal,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Transect_Sal = subset(dat_Transect_Sal, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Transect_Sal = rbind(Nonsignificant_df_Transect_Sal, Significant_df_Transect_Sal)

#Plot
plotme_Transect_Sal = ggplot(plotsign.vs.nonSign_Transect_Sal,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Transect_Sal

dev.copy(png, "Sign.vs.NonSign_Transect_Sal", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Transect_Sal$Genus))

#Total number of sample
 40043       +     16468
#56511

#Ratio of significant
16468/56511
#29.1

#Ratio of non-significant
40043/56511
#70.9

```
##Correlated with temperature
```{r Set up horizontal temperature correlation}
#Group them by their Temperature
dge_EdgeR_obj_Transect_Temp = phyloseq_to_edgeR(Fiord_phyloseq_Horonly, group = "Temperature")

# Perform binary test
et_EdgeR_Transect_Temp = exactTest(dge_EdgeR_obj_Transect_Temp)
# Extract values from test results
tt_EdgeR_Transect_Temp = topTags(et_EdgeR_Transect_Temp, n = nrow(dge_EdgeR_obj_Transect_Temp$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Transect_Temp = tt_EdgeR_Transect_Temp@.Data[[1]]
sigtab_2fold_EdgeR_Transect_Temp<- subset(res_EdgeR_Transect_Temp, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1
sigtab_2fold_FDR_Transect_Temp <- subset(sigtab_2fold_EdgeR_Transect_Temp, FDR < 0.1)

keepTaxa_FDR_Transect_Temp <- sigtab_2fold_EdgeR_Transect_Temp$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Transect_Temp <- subset_taxa(Fiord_phyloseq_Horonly, Genus %in% keepTaxa_FDR_Transect_Temp) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Transect_Temp <- tax_glom(Twofold_FDR_Transect_Temp, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Transect_Temp <- dat_2fold_FDR_Transect_Temp[order(dat_2fold_FDR_Transect_Temp$Family),] #Order them at the Phylum level

dat_2fold_FDR_Transect_Temp$Phylum <- as.character(dat_2fold_FDR_Transect_Temp$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Transect_Temp <- ddply(dat_2fold_FDR_Transect_Temp, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Transect_Temp <- medians_Transect_Temp[medians_Transect_Temp$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Transect_Temp[dat_2fold_FDR_Transect_Temp$Phylum %in% remainder_Transect_Temp,]$Phylum <- 'RareTaxa'

Summary_Transect_Temp <- summarySE(dat_2fold_FDR_Transect_Temp, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "Temperature", "Distance_from_L1"))
Summary_Transect_Temp

write.csv(Summary_Transect_Temp, file = "Summary_for_Nadjejda_Transect_16S_Temp.csv")

Summary_Transect_Temp<-dplyr::arrange(Summary_Transect_Temp,Phylum, Abundance)

Summary_Transect_Temp$Phylum <- factor(Summary_Transect_Temp$Phylum,
                         levels=(unique(Summary_Transect_Temp$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Transect_Temp$Phylum)

#Rename levels
levels(Summary_Transect_Temp$Phylum) = c("Bacteroidetes",
                                         "Cyanobacteria",
                                         "Proteobacteria",
                                         "Rare Taxa (<1%)")

#Check to make sure it worked
levels(Summary_Transect_Temp$Phylum) 

#Save as table for ease of reading.
write.csv(Summary_Transect_Temp, file = "Ordered_summary_Transect_phylalvl_Temp.csv")
```
```{r Plot significantly correlated horizontal temperature community}
Summary_plot_Transect_Temp<-ggplot(Summary_Transect_Temp, 
                        aes(x=as.numeric(as.character(Distance_from_L1)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Km from the outermost sample")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)+
  facet_wrap(~Sample_Depth)


Summary_plot_Transect_Temp


```
```{r Significant vs. non-significant community - Horizontal temperature}
# get abundance in %
Sample_counts_Transect_Temp <- transform_sample_counts(Fiord_phyloseq_Horonly, function(x) x/sum(x))

# agglomerate taxa
glom_Transect_Temp <- tax_glom(Sample_counts_Transect_Temp, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Transect_Temp <- psmelt(glom_Transect_Temp)


# convert Phylum to a character vector from a factor because R
dat_Transect_Temp$Genus <- as.character(dat_Transect_Temp$Genus)

#Convert to character, as R
Significant_taxa_Transect_Temp <- as.character(keepTaxa_FDR_Transect_Temp)

#Change the name of the significant organisms
dat_Transect_Temp[dat_Transect_Temp$Genus %in% Significant_taxa_Transect_Temp,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Transect_Temp = subset(dat_Transect_Temp, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Transect_Temp[dat_Transect_Temp$Genus %!in% Significant_df_Transect_Temp,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Transect_Temp = subset(dat_Transect_Temp, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Transect_Temp = rbind(Nonsignificant_df_Transect_Temp, Significant_df_Transect_Temp)

#Plot
plotme_Transect_Temp = ggplot(plotsign.vs.nonSign_Transect_Temp,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Transect_Temp

dev.copy(png, "Sign.vs.NonSign_Transect_Temp", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Transect_Temp$Genus))

#Total number of sample
 40043       +     16468
#56511

#Ratio of significant
16468/56511
#29.1

#Ratio of non-significant
40043/56511
#70.9


```
##Correlated with oxygen
```{r Set up horizontal oxygen correlation}
#Group them by their OxygenPerSat
dge_EdgeR_obj_Transect_Ox = phyloseq_to_edgeR(Fiord_phyloseq_Horonly, group = "OxygenPerSat")

# Perform binary test
et_EdgeR_Transect_Ox = exactTest(dge_EdgeR_obj_Transect_Ox)
# Extract values from test results
tt_EdgeR_Transect_Ox = topTags(et_EdgeR_Transect_Ox, n = nrow(dge_EdgeR_obj_Transect_Ox$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Transect_Ox = tt_EdgeR_Transect_Ox@.Data[[1]]
sigtab_2fold_EdgeR_Transect_Ox<- subset(res_EdgeR_Transect_Ox, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1
sigtab_2fold_FDR_Transect_Ox <- subset(sigtab_2fold_EdgeR_Transect_Ox, FDR < 0.1)

keepTaxa_FDR_Transect_Ox <- sigtab_2fold_EdgeR_Transect_Ox$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Transect_Ox <- subset_taxa(Fiord_phyloseq_Horonly, Genus %in% keepTaxa_FDR_Transect_Ox) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Transect_Ox <- tax_glom(Twofold_FDR_Transect_Ox, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Transect_Ox <- dat_2fold_FDR_Transect_Ox[order(dat_2fold_FDR_Transect_Ox$Family),] #Order them at the Phylum level

dat_2fold_FDR_Transect_Ox$Phylum <- as.character(dat_2fold_FDR_Transect_Ox$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Transect_Ox <- ddply(dat_2fold_FDR_Transect_Ox, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Transect_Ox <- medians_Transect_Ox[medians_Transect_Ox$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Transect_Ox[dat_2fold_FDR_Transect_Ox$Phylum %in% remainder_Transect_Ox,]$Phylum <- 'RareTaxa'

Summary_Transect_Ox <- summarySE(dat_2fold_FDR_Transect_Ox, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "OxygenPerSat", "Distance_from_L1"))
Summary_Transect_Ox

write.csv(Summary_Transect_Ox, file = "Summary_for_Nadjejda_Transect_16S_Ox.csv")

Summary_Transect_Ox<-dplyr::arrange(Summary_Transect_Ox,Phylum, Abundance)

Summary_Transect_Ox$Phylum <- factor(Summary_Transect_Ox$Phylum,
                         levels=(unique(Summary_Transect_Ox$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Transect_Ox$Phylum)

#Rename levels
levels(Summary_Transect_Ox$Phylum) = c("Actinobacteria",
                                         "Bacteroidetes",
                                         "Cyanobacteria",
                                         "Marinimicrobia",
                                         "Planctomycetes",
                                         "Proteobacteria",
                                         "Rare Taxa (<1%)")

#Check to make sure it worked
levels(Summary_Transect_Ox$Phylum) 

#Save as table for ease of reading.
write.csv(Summary_Transect_Ox, file = "Ordered_summary_Transect_phylalvl_Ox.csv")

```
```{r Plot significantly correlated horizontal oxygen community}
Summary_plot_Transect_Ox<-ggplot(Summary_Transect_Ox, 
                        aes(x=as.numeric(as.character(Distance_from_L1)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Km from the outermost sample")+
  ylab("Mean relative abundance")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)+
  facet_wrap(~Sample_Depth)


Summary_plot_Transect_Ox

```
```{r Significant vs. non-significant community - Horizontal oxygen}
# get abundance in %
Sample_counts_Transect_Ox <- transform_sample_counts(Fiord_phyloseq_Horonly, function(x) x/sum(x))

# agglomerate taxa
glom_Transect_Ox <- tax_glom(Sample_counts_Transect_Ox, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Transect_Ox <- psmelt(glom_Transect_Ox)


# convert Phylum to a character vector from a factor because R
dat_Transect_Ox$Genus <- as.character(dat_Transect_Ox$Genus)

#Convert to character, as R
Significant_taxa_Transect_Ox <- as.character(keepTaxa_FDR_Transect_Ox)

#Change the name of the significant organisms
dat_Transect_Ox[dat_Transect_Ox$Genus %in% Significant_taxa_Transect_Ox,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Transect_Ox = subset(dat_Transect_Ox, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Transect_Ox[dat_Transect_Ox$Genus %!in% Significant_df_Transect_Ox,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Transect_Ox = subset(dat_Transect_Ox, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Transect_Ox = rbind(Nonsignificant_df_Transect_Ox, Significant_df_Transect_Ox)

#Plot
plotme_Transect_Ox = ggplot(plotsign.vs.nonSign_Transect_Ox,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Transect_Ox

dev.copy(png, "Sign.vs.NonSign_Transect_Ox", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Transect_Ox$Genus))

#Total number of sample
 40043       +     7337
#47380

#Ratio of significant
7337/47380
#15.5

#Ratio of non-significant
40043/47380
#84.5


```
##Export for ggarrange
```{r Export horizontal}
saveRDS(Summary_plot_Transect_Dist,"~/Desktop/Phylum_EdgeR_Transect_16S_Dist.rds")
saveRDS(Summary_plot_Transect_Sal,"~/Desktop/Phylum_EdgeR_Transect_16S_Sal.rds")
saveRDS(Summary_plot_Transect_Temp,"~/Desktop/Phylum_EdgeR_Transect_16S_Temp.rds")
saveRDS(Summary_plot_Transect_Ox,"~/Desktop/Phylum_EdgeR_Transect_16S_Ox.rds")
```
#Signficantly changing vertical community
##Correlated with depth
```{r Set up vertical depth correlation}


#Group them by their depths
dge_EdgeR_obj = phyloseq_to_edgeR(Fiord_phyloseq_Vert, group = "Sample_Depth")

# Perform binary test
et_EdgeR = exactTest(dge_EdgeR_obj)
# Extract values from test results
tt_EdgeR = topTags(et_EdgeR, n = nrow(dge_EdgeR_obj$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR = tt_EdgeR@.Data[[1]]
sigtab_2fold_EdgeR<- subset(res_EdgeR, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1 - as there are none - have to use something else
sigtab_2fold_FDR <- subset(sigtab_2fold_EdgeR, FDR < 0.1)

library(dplyr)
keepTaxa_FDR <- sigtab_2fold_EdgeR$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR <- subset_taxa(Fiord_phyloseq_Vert, Genus %in% keepTaxa_FDR) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR <- tax_glom(Twofold_FDR, taxrank = 'Family') %>%#Merge the species at the Family level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR <- dat_2fold_FDR[order(dat_2fold_FDR$Family),] #Order them at the Family level

dat_2fold_FDR$Phylum <- as.character(dat_2fold_FDR$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians <- ddply(dat_2fold_FDR, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder <- medians[medians$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR[dat_2fold_FDR$Phylum %in% remainder,]$Phylum <- 'RareTaxa'

Summary <- summarySE(dat_2fold_FDR, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth"))
Summary

write.csv(Summary, file = "Summary_for_Nadjejda_16S_Depth.csv") 

Summary<-dplyr::arrange(Summary,Phylum, Abundance)

Summary$Phylum <- factor(Summary$Phylum,
                         levels=(unique(Summary$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary$Phylum)

#Rename levels
levels(Summary$Phylum) = c("Acidobacteria",
                           "Actinobacteria", 
                   "Bacteroidetes",
                   "Chloroflexi",
                   "Cyanobacteria",
                   "Euryarchaeota",
                   "Marinimicrobia",
                   "Planctomycetes",
                   "Proteobacteria",
                   "Rare Taxa (<1%)",
                   "Thaumarchaeota",
                   "Verrucomicrobia")

levels(Summary$Phylum) #Check to make sure it worked

#Save as table for ease of reading.
write.csv(Summary, file = "Ordered_summary_Depth.csv")

```
```{r Plot significantly correlated vertical depth community}
Summary_plot<-ggplot(Summary, 
                        aes(x=as.numeric(as.character(Sample_Depth)), 
                            y=Abundance*100, 
                            colour=Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("")+
  ylab("")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)

```
```{r Significant vs. non-significant community - vertical depth}
# get abundance in %
Sample_counts <- transform_sample_counts(Fiord_phyloseq_Vert, function(x) x/sum(x))

# agglomerate taxa
glom <- tax_glom(Sample_counts, taxrank = 'Genus')

# create dataframe from phyloseq object
dat <- psmelt(glom)


# convert Phylum to a character vector from a factor because R
dat$Genus <- as.character(dat$Genus)

#Convert to character, as R
Significant_taxa <- as.character(keepTaxa_FDR)

#Change the name of the significant organisms
dat[dat$Genus %in% Significant_taxa,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df = subset(dat, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat[dat$Genus %!in% Significant_df,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df = subset(dat, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign = rbind(Nonsignificant_df, Significant_df)

#Plot
plotme = ggplot(plotsign.vs.nonSign,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme

dev.copy(png, "Sign.vs.NonSign", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign$Genus))

#Total number of sample
 20892       +     7344
#28236

#Ratio of significant
7344/28236
#26%

#Ratio of non-significant
20892/28236
#74%

```
##Correlated with salinity
```{r Set up vertical salinity  correlation}
#Create CTD df
CTDtoadd = data.frame(Sample_Depth = c("0","10","40", "100", "200", "360"), OxygenPerSat = as.numeric(0), Salinity = as.numeric(0), Temperature = as.numeric(0))
#Expand df to fit all genera.
CTDtoadd$OxygenPerSat = c(7.919300, 6.930200, 6.016600, 6.207767, 6.683000, 6.506733)
CTDtoadd$Salinity = c(29.17795, 33.83610, 34.12873, 34.12710, 34.12883, 34.13773)
CTDtoadd$Temperature = c(2.69155, 12.10170, 12.25490, 12.08940, 11.97240, 11.99627)

#Merge CTD df and smaple data
CTD_df = cbind(CTDtoadd, Fiord_phyloseq_Vert@sam_data)

#Replace the old sample data with the new df
Fiord_phyloseq_Vert@sam_data = sample_data(CTD_df)

#Group them by their depths
dge_EdgeR_obj_Sal = phyloseq_to_edgeR(Fiord_phyloseq_Vert, group = "Salinity")

# Perform binary test
et_EdgeR_Sal = exactTest(dge_EdgeR_obj_Sal)
# Extract values from test results
tt_EdgeR_Sal = topTags(et_EdgeR_Sal, n = nrow(dge_EdgeR_obj_Sal$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Sal = tt_EdgeR_Sal@.Data[[1]]
sigtab_2fold_EdgeR_Sal<- subset(res_EdgeR_Sal, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1 - as there are none - have to use something else
sigtab_2fold_FDR_Sal <- subset(sigtab_2fold_EdgeR_Sal, FDR < 0.1)

keepTaxa_FDR_Sal <- sigtab_2fold_EdgeR_Sal$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Sal <- subset_taxa(Fiord_phyloseq_Vert, Genus %in% keepTaxa_FDR_Sal) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Sal <- tax_glom(Twofold_FDR_Sal, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Sal <- dat_2fold_FDR_Sal[order(dat_2fold_FDR_Sal$Family),] #Order them at the Phylum level

dat_2fold_FDR_Sal$Phylum <- as.character(dat_2fold_FDR_Sal$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Sal <- ddply(dat_2fold_FDR_Sal, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Sal <- medians_Sal[medians_Sal$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Sal[dat_2fold_FDR_Sal$Phylum %in% remainder_Sal,]$Phylum <- 'RareTaxa'

Summary_Sal <- summarySE(dat_2fold_FDR_Sal, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "Salinity"))
Summary_Sal

write.csv(Summary_Sal, file = "Summary_for_Nadjejda_Profile_16S_Sal.csv")

Summary_Sal<-dplyr::arrange(Summary_Sal,Phylum, Abundance)

Summary_Sal$Phylum <- factor(Summary_Sal$Phylum,
                         levels=(unique(Summary_Sal$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Sal$Phylum)

#Rename levels
levels(Summary_Sal$Phylum) = c("Acidobacteria",
                           "Actinobacteria", 
                   "Bacteroidetes",
                   "Chloroflexi",
                   "Cyanobacteria",
                   "Euryarchaeota",
                   "Marinimicrobia",
                   "Planctomycetes",
                   "Proteobacteria",
                   "Rare Taxa (<1%)",
                   "Thaumarchaeota",
                   "Verrucomicrobia")

levels(Summary_Sal$Phylum) #Check to make sure it worked

#Save as table for ease of reading.
write.csv(Summary_Sal, file = "Ordered_summary_Sal.csv")
```
```{r Plot significantly correlated vertical salinity community}
Summary_plot_Sal<-ggplot(Summary_Sal, 
                        aes(x=as.factor(as.numeric(as.character(Sample_Depth))), 
                            y=Abundance*100, 
                            colour=Phylum,
                            group = Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("")+
  ylab("")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)

```
```{r Significant vs. non-significant community - vertical salinity}
# get abundance in %
Sample_counts_Sal <- transform_sample_counts(Fiord_phyloseq_Vert, function(x) x/sum(x))

# agglomerate taxa
glom_Sal <- tax_glom(Sample_counts_Sal, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Sal <- psmelt(glom_Sal)


# convert Phylum to a character vector from a factor because R
dat_Sal$Genus <- as.character(dat_Sal$Genus)

#Convert to character, as R
Significant_taxa_Sal <- as.character(keepTaxa_FDR_Sal)

#Change the name of the significant organisms
dat_Sal[dat_Sal$Genus %in% Significant_taxa_Sal,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Sal = subset(dat_Sal, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Sal[dat_Sal$Genus %!in% Significant_df_Sal,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Sal = subset(dat_Sal, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Sal = rbind(Nonsignificant_df_Sal, Significant_df_Sal)

#Plot
plotme_Sal = ggplot(plotsign.vs.nonSign_Sal,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Sal

dev.copy(png, "Sign.vs.NonSign_Sal", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Sal$Genus))

#Total number of sample
 20892       +     7392
#28284

#Ratio of significant
7392/28284
#26.1%

#Ratio of non-significant
20892/28284
#73.9%
```
##Correlated with temperature
```{r Set up vertical temperature correlation}
#Create CTD df
CTDtoadd = data.frame(Sample_Depth = c("0","10","40", "100", "200", "360"), OxygenPerSat = as.numeric(0), Salinity = as.numeric(0), Temperature = as.numeric(0))
#Expand df to fit all genera.
CTDtoadd$OxygenPerSat = c(7.919300, 6.930200, 6.016600, 6.207767, 6.683000, 6.506733)
CTDtoadd$Salinity = c(29.17795, 33.83610, 34.12873, 34.12710, 34.12883, 34.13773)
CTDtoadd$Temperature = c(2.69155, 12.10170, 12.25490, 12.08940, 11.97240, 11.99627)

#Merge CTD df and smaple data
CTD_df = cbind(CTDtoadd, Fiord_phyloseq_Vert@sam_data)

#Replace the old sample data with the new df
Fiord_phyloseq_Vert@sam_data = sample_data(CTD_df)

#Group them by their depths
dge_EdgeR_obj_Temp = phyloseq_to_edgeR(Fiord_phyloseq_Vert, group = "Temperature")

# Perform binary test
et_EdgeR_Temp = exactTest(dge_EdgeR_obj_Temp)
# Extract values from test results
tt_EdgeR_Temp = topTags(et_EdgeR_Temp, n = nrow(dge_EdgeR_obj_Temp$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Temp = tt_EdgeR_Temp@.Data[[1]]
sigtab_2fold_EdgeR_Temp<- subset(res_EdgeR_Temp, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1 - as there are none - have to use something else
sigtab_2fold_FDR_Temp <- subset(sigtab_2fold_EdgeR_Temp, FDR < 0.1)

keepTaxa_FDR_Temp <- sigtab_2fold_EdgeR_Temp$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Temp <- subset_taxa(Fiord_phyloseq_Vert, Genus %in% keepTaxa_FDR_Temp) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Temp <- tax_glom(Twofold_FDR_Temp, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Temp <- dat_2fold_FDR_Temp[order(dat_2fold_FDR_Temp$Family),] #Order them at the Phylum level

dat_2fold_FDR_Temp$Phylum <- as.character(dat_2fold_FDR_Temp$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Temp <- ddply(dat_2fold_FDR_Temp, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Temp <- medians_Temp[medians_Temp$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Temp[dat_2fold_FDR_Temp$Phylum %in% remainder_Temp,]$Phylum <- 'RareTaxa'

Summary_Temp <- summarySE(dat_2fold_FDR_Temp, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "Temperature"))
Summary_Temp

write.csv(Summary_Temp, file = "Summary_for_Nadjejda_Profile_16S_Temp.csv")

Summary_Temp<-dplyr::arrange(Summary_Temp,Phylum, Abundance)

Summary_Temp$Phylum <- factor(Summary_Temp$Phylum,
                         levels=(unique(Summary_Temp$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Temp$Phylum)

#Rename levels
levels(Summary_Temp$Phylum) = c("Acidobacteria",
                           "Actinobacteria", 
                   "Bacteroidetes",
                   "Chloroflexi",
                   "Cyanobacteria",
                   "Euryarchaeota",
                   "Marinimicrobia",
                   "Planctomycetes",
                   "Proteobacteria",
                   "Rare Taxa (<1%)",
                   "Thaumarchaeota",
                   "Verrucomicrobia")

levels(Summary_Temp$Phylum) #Check to make sure it worked

#Save as table for ease of reading.
write.csv(Summary_Temp, file = "Ordered_summary_Temp.csv")

```
```{r Plot significantly correlated vertical temperature community}
Summary_plot_Temp<-ggplot(Summary_Temp, 
                        aes(x=as.factor(as.numeric(as.character(Sample_Depth))), 
                            y=Abundance*100, 
                            colour=Phylum,
                            group = Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("")+
  ylab("")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)

```
```{r Significant vs. non-significant community - vertical temperature}
# get abundance in %
Sample_counts_Temp <- transform_sample_counts(Fiord_phyloseq_Vert, function(x) x/sum(x))

# agglomerate taxa
glom_Temp <- tax_glom(Sample_counts_Temp, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Temp <- psmelt(glom_Temp)


# convert Phylum to a character vector from a factor because R
dat_Temp$Genus <- as.character(dat_Temp$Genus)

#Convert to character, as R
Significant_taxa_Temp <- as.character(keepTaxa_FDR_Temp)

#Change the name of the significant organisms
dat_Temp[dat_Temp$Genus %in% Significant_taxa_Temp,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Temp = subset(dat_Temp, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Temp[dat_Temp$Genus %!in% Significant_df_Temp,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Temp = subset(dat_Temp, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Temp = rbind(Nonsignificant_df_Temp, Significant_df_Temp)

#Plot
plotme_Temp = ggplot(plotsign.vs.nonSign_Temp,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Temp

dev.copy(png, "Sign.vs.NonSign_Temp", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Temp$Genus))

#Total number of sample
 20892       +     7296
#28188

#Ratio of significant
7296/28188
#25.9%

#Ratio of non-significant
20892/28188
#74.1

```
##Correlated with oxygen
```{r Set up vertical oxygen correlation}
#Create CTD df
CTDtoadd = data.frame(Sample_Depth = c("0","10","40", "100", "200", "360"), OxygenPerSat = as.numeric(0), Salinity = as.numeric(0), Temperature = as.numeric(0))
#Expand df to fit all genera.
CTDtoadd$OxygenPerSat = c(7.919300, 6.930200, 6.016600, 6.207767, 6.683000, 6.506733)
CTDtoadd$Salinity = c(29.17795, 33.83610, 34.12873, 34.12710, 34.12883, 34.13773)
CTDtoadd$Temperature = c(2.69155, 12.10170, 12.25490, 12.08940, 11.97240, 11.99627)

#Merge CTD df and sample data
CTD_df = cbind(CTDtoadd, Fiord_phyloseq_Vert@sam_data)

#Replace the old sample data with the new df
Fiord_phyloseq_Vert@sam_data = sample_data(CTD_df)

#Group them by their depths
dge_EdgeR_obj_Ox = phyloseq_to_edgeR(Fiord_phyloseq_Vert, group = "OxygenPerSat")

# Perform binary test
et_EdgeR_Ox = exactTest(dge_EdgeR_obj_Ox)
# Extract values from test results
tt_EdgeR_Ox = topTags(et_EdgeR_Ox, n = nrow(dge_EdgeR_obj_Ox$table), adjust.method = "BH", sort.by = "PValue")
res_EdgeR_Ox = tt_EdgeR_Ox@.Data[[1]]
sigtab_2fold_EdgeR_Ox<- subset(res_EdgeR_Ox, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)

#Keep only FDR corrected <.1 - as there are none - have to use something else
sigtab_2fold_FDR_Ox <- subset(sigtab_2fold_EdgeR_Ox, FDR < 0.1)


keepTaxa_FDR_Ox <- sigtab_2fold_EdgeR_Ox$Genus #Extract the OTU table that was shown to be significant
Twofold_FDR_Ox <- subset_taxa(Fiord_phyloseq_Vert, Genus %in% keepTaxa_FDR_Ox) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_Ox <- tax_glom(Twofold_FDR_Ox, taxrank = 'Family') %>%#Merge the species at the Phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_Ox <- dat_2fold_FDR_Ox[order(dat_2fold_FDR_Ox$Family),] #Order them at the Phylum level

dat_2fold_FDR_Ox$Phylum <- as.character(dat_2fold_FDR_Ox$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_Ox <- ddply(dat_2fold_FDR_Ox, ~Phylum, function(x) c(median=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_Ox <- medians_Ox[medians_Ox$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_Ox[dat_2fold_FDR_Ox$Phylum %in% remainder_Ox,]$Phylum <- 'RareTaxa'

Summary_Ox <- summarySE(dat_2fold_FDR_Ox, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_Depth", "OxygenPerSat"))
Summary_Ox

write.csv(Summary_Ox, file = "Summary_for_Nadjejda_Profile_16S_Ox.csv")

Summary_Ox<-dplyr::arrange(Summary_Ox,Phylum, Abundance)

Summary_Ox$Phylum <- factor(Summary_Ox$Phylum,
                         levels=(unique(Summary_Ox$Phylum)))

#Check for weird names and make Rare taxa name better.
levels(Summary_Ox$Phylum)

#Rename levels
levels(Summary_Ox$Phylum) = c("Acidobacteria",
                           "Actinobacteria", 
                   "Bacteroidetes",
                   "Chloroflexi",
                   "Cyanobacteria",
                   "Euryarchaeota",
                   "Marinimicrobia",
                   "Planctomycetes",
                   "Proteobacteria",
                   "Rare Taxa (<1%)",
                   "Thaumarchaeota",
                   "Verrucomicrobia")

levels(Summary_Ox$Phylum) #Check to make sure it worked

#Save as table for ease of reading.
write.csv(Summary_Ox, file = "Ordered_summary_Ox.csv")

```
```{r Plot significantly correlated vertical oxygen community}

Summary_plot_Ox<-ggplot(Summary_Ox, 
                        aes(x=as.factor(as.numeric(as.character(Sample_Depth))), 
                            y=Abundance*100, 
                            colour=Phylum,
                            group = Phylum))+ 
  geom_line(stat = "identity") + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black", 
                position=pd
                )+
  theme_bw()+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("")+
  ylab("")+
  scale_colour_manual("Phylum", values = Phylum_colour_list)

```
```{r Significant vs. non-significant community - vertical oxygen}
# get abundance in %
Sample_counts_Ox <- transform_sample_counts(Fiord_phyloseq_Vert, function(x) x/sum(x))

# agglomerate taxa
glom_Ox <- tax_glom(Sample_counts_Ox, taxrank = 'Genus')

# create dataframe from phyloseq object
dat_Ox <- psmelt(glom_Ox)


# convert Phylum to a character vector from a factor because R
dat_Ox$Genus <- as.character(dat_Ox$Genus)

#Convert to character, as R
Significant_taxa_Ox <- as.character(keepTaxa_FDR_Ox)

#Change the name of the significant organisms
dat_Ox[dat_Ox$Genus %in% Significant_taxa_Ox,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df_Ox = subset(dat_Ox, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat_Ox[dat_Ox$Genus %!in% Significant_df_Ox,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df_Ox = subset(dat_Ox, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign_Ox = rbind(Nonsignificant_df_Ox, Significant_df_Ox)

#Plot
plotme_Ox = ggplot(plotsign.vs.nonSign_Ox,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme_Ox

dev.copy(png, "Sign.vs.NonSign_Ox", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign_Ox$Genus))

#Total number of sample
 20892       +     7344
#28236

#Ratio of significant
7344/28236
#26%

#Ratio of non-significant
20892/28236
#74%

```
##Export for ggarrange
```{r Export vertical}
saveRDS(Summary_plot,"~/Desktop/Fjordpaper/3_Results/11_To_ggarrange/Phylum_EdgeR_Profile_16S_Depth.rds")
saveRDS(Summary_plot_Ox,"~/Desktop/Fjordpaper/3_Results/11_To_ggarrange/Phylum_EdgeR_Profile_16S_Ox.rds")
saveRDS(Summary_plot_Sal,"~/Desktop/Fjordpaper/3_Results/11_To_ggarrange/Phylum_EdgeR_Profile_16S_Sal.rds")
saveRDS(Summary_plot_Temp,"~/Desktop/Fjordpaper/3_Results/11_To_ggarrange/Phylum_EdgeR_Profile_16S_Temp.rds")
```
#Signficantly changing consensus community
```{r Significant changes inside Fjord between depths}
#Do exact test for Sample types starting with in

#Subset to in with different depths
Fiord_phyloseq_in = subset_samples(Fiord_phyloseq, Sample_type == "in")

#Group them by their depths
dge_D1S1_depthin = phyloseq_to_edgeR(Fiord_phyloseq_in, group = "Sample_Depth")

# Perform binary test
et_D1S1_depthin = exactTest(dge_D1S1_depthin)
# Extract values from test results
tt_D1S1_depthin = topTags(et_D1S1_depthin, n = nrow(dge_D1S1_depthin$table), adjust.method = "BH", sort.by = "PValue")
res_D1S1_depthin = tt_D1S1_depthin@.Data[[1]]
sigtab_2fold_D1S1_depthin <- subset(res_D1S1_depthin, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)
sigtab_2fold_D1S1_depthin$Sample_type<-c("in")

```
```{r Significant changes outside Fjord between depths}

#Repeat for out
#Subset to in with different depths
Fiord_phyloseq_out = subset_samples(Fiord_phyloseq, Sample_type == "out")

#Group them by their depths - as this is the difference we are assessing.
dge_D1S1_depthout = phyloseq_to_edgeR(Fiord_phyloseq_out, group = "Sample_Depth")

# Perform boutary test
et_D1S1_depthout = exactTest(dge_D1S1_depthout)
# Extract values from test results
tt_D1S1_depthout = topTags(et_D1S1_depthout, n = nrow(dge_D1S1_depthout$table), adjust.method = "BH", sort.by = "PValue")
res_D1S1_depthout = tt_D1S1_depthout@.Data[[1]]
sigtab_2fold_D1S1_depthout <- subset(res_D1S1_depthout, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)
sigtab_2fold_D1S1_depthout$Sample_type<-c("out")
```
```{r Significant changes at the surface between inner and outer FJord regions}

#Repeat for surface
##Subset to surface samples
Fiord_phyloseq_surface = subset_samples(Fiord_phyloseq, Sample_Depth == "0")

#Group them by their depths - as this is the difference we are assessing.
dge_D1S1_typesurface = phyloseq_to_edgeR(Fiord_phyloseq_surface, group = "Sample_type")

# Perform boundary test
et_D1S1_typesurface = exactTest(dge_D1S1_typesurface)
# Extract values from test results
tt_D1S1_typesurface = topTags(et_D1S1_typesurface, n = nrow(dge_D1S1_typesurface$table), adjust.method = "BH", sort.by = "PValue")
res_D1S1_typesurface = tt_D1S1_typesurface@.Data[[1]]
sigtab_2fold_D1S1_typesurface <- subset(res_D1S1_typesurface, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)
sigtab_2fold_D1S1_typesurface$Sample_Depth<-c("0")

```
```{r Significant changes at 10 m between inner and outer FJord regions}

#Repeat with 10m samples
##Subset to 10m samples
Fiord_phyloseq_10m = subset_samples(Fiord_phyloseq, Sample_Depth == "10")

##Group them by their depths - as this is the difference we are assessing.
dge_D1S1_type10m = phyloseq_to_edgeR(Fiord_phyloseq_10m, group = "Sample_type")

## Perform boundary test
et_D1S1_type10m = exactTest(dge_D1S1_type10m)
# Extract values from test results
tt_D1S1_type10m = topTags(et_D1S1_type10m, n = nrow(dge_D1S1_type10m$table), adjust.method = "BH", sort.by = "PValue")
res_D1S1_type10m = tt_D1S1_type10m@.Data[[1]]
sigtab_2fold_D1S1_type10m <- subset(res_D1S1_type10m, PValue < 0.05 & logFC >= 2 | PValue < 0.05 & logFC <= -2)
sigtab_2fold_D1S1_type10m$Sample_Depth<-c("10")

```
```{r Set up for depth and region correlation for consensus fjord}
#Keep only FDR corrected <.1
sigtab_2fold_surface_FDR <- subset(sigtab_2fold_D1S1_typesurface, FDR < 0.1)
sigtab_2fold_10m_FDR <- subset(sigtab_2fold_D1S1_type10m, FDR < 0.1)
sigtab_2fold_in_FDR <- subset(sigtab_2fold_D1S1_depthin, FDR < 0.1)
sigtab_2fold_out_FDR <- subset(sigtab_2fold_D1S1_depthout, FDR < 0.1)

keepTaxa_FDR_all <- c(as.character(sigtab_2fold_surface_FDR$OTU),as.character(sigtab_2fold_10m_FDR$OTU),as.character(sigtab_2fold_in_FDR$OTU),as.character(sigtab_2fold_out_FDR$OTU))  #Extract the OTU table that was shown to be significant

Twofold_FDR_all <- subset_taxa(Fiord_phyloseq_inout, OTU %in% keepTaxa_FDR_all) #Subset the taxa by the OTUs that were shown to change significantly
dat_2fold_FDR_all <- tax_glom(Twofold_FDR_all, taxrank = 'Family') %>%#Merge the species at the phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #Extract abundance data from phyloseq object
  psmelt() #Melt it into a dataframe
dat_2fold_FDR_all <- dat_2fold_FDR_all[order(dat_2fold_FDR_all$Family),] #Order them at the Phylum level

dat_2fold_FDR_all$Phylum <- as.character(dat_2fold_FDR_all$Phylum)
  # group dataframe by Phylum, calculate relative abundance
medians_all <- ddply(dat_2fold_FDR_all, ~Phylum, function(x) c(medians_all=mean(x$Abundance)))
  # find Phyla whose rel. abund. is less than 1%
remainder_all <- medians_all[medians_all$median <= 0.01,]$Phylum
  # change their name to "Remainder"
dat_2fold_FDR_all[dat_2fold_FDR_all$Phylum %in% remainder_all,]$Phylum <- 'RareTaxa'

Summary_all<- summarySE(dat_2fold_FDR_all, measurevar="Abundance", groupvars=c("Domain", "Phylum", "Class", "Order", "Family", "Sample_type", "Sample_Depth"))
Summary_all


#check naming of Genera to make sure they are informative otherwise save, fix and replace
write.csv(Summary_all, "Summary_for_Nadjejda_Multi.csv")

 

Summary_all<-dplyr::arrange(Summary_all,Phylum, Abundance)

Summary_all$Phylum <- factor(Summary_all$Phylum,
                                         levels=(unique(Summary_all$Phylum)))

Summary_all$Sample_type<-factor(Summary_all$Sample_type, levels=c("in","out"))
Summary_all$Sample_Depth<-factor(Summary_all$Sample_Depth, levels=c("0","10"))

#Save as table for ease of reading.
write.csv(Summary_all, file = "Ordered_summary_all.csv")

#Change the facet wrap labels
levels(Summary_all$Sample_Depth)
levels(Summary_all$Sample_Depth) = c("0 m", "10 m")

levels(Summary_all$Sample_type)
levels(Summary_all$Sample_type) = c("Inner", "Outer")
```
```{r Plot significantly correlated consensus fjord community}

Summary_all_plot<-ggplot(Summary_all, 
                        aes(x=Phylum, 
                            y=Abundance*100, 
                            fill=Phylum))+ 
  geom_bar(stat = "identity", width = 0.85) + 
  geom_errorbar(aes(ymin=(Abundance-se)*100, 
                    ymax=(Abundance+se)*100), 
                colour="black"#, 
                #position=pd
                )+
  theme_bw() + 
  facet_grid(Sample_type~Sample_Depth, 
             drop = TRUE, 
             space = "fixed", 
             scales = "free", 
             margins = "TRUE", 
             labeller = label_wrap_gen(width = 16, multi_line = TRUE))+
  theme(legend.position = "right", 
        strip.background = element_blank())+
  xlab("Phylum")+
  ylab("Mean relative abundance (%)")+
  scale_fill_manual("Phylum", values = Phylum_colour_list,  labels= c("Actinobacteria", "Bacteroidetes", "Cyanobacteria", "Euryarchaeota","Planctomycetes", "Proteobacteria", "Rare taxa (<1%)", "Verrucomicrobia"))+
  theme(axis.text.x =element_blank())

Summary_all_plot
dev.copy(png, "Significant_phylum_all", units = "in", width = 5, height = 4, res = 500)
dev.off()

```
```{r Significant vs. non-significant community - consensus depth and region}
# get abundance in %
Sample_counts <- transform_sample_counts(Fiord_phyloseq_inout, function(x) x/sum(x))

# agglomerate taxa
glom <- tax_glom(Sample_counts, taxrank = 'Genus')

# create dataframe from phyloseq object
dat <- psmelt(glom)


# convert Phylum to a character vector from a factor because R
dat$Genus <- as.character(dat$Genus)

#Convert to character, because R
Significant_taxa <- c(as.character(sigtab_2fold_surface_FDR$Genus),as.character(sigtab_2fold_10m_FDR$Genus),as.character(sigtab_2fold_in_FDR$Genus),as.character(sigtab_2fold_out_FDR$Genus))  #Extract the OTU table that was shown to be significant

#Change the name of the significant organisms
dat[dat$Genus %in% Significant_taxa,]$Genus <- 'Significant'

#Extract the significant organisms into a new df
Significant_df = subset(dat, Genus == "Significant")

#Make a function the opposite of %in%
'%!in%' <- function(x,y)!('%in%'(x,y))

#Rename the non-significant organisms
dat[dat$Genus %!in% Significant_taxa,]$Genus <- 'Nonsignificant'

#Extract the non-significant organisms
Nonsignificant_df = subset(dat, Genus == "Nonsignificant")

#Merge the significant and non-significant dfs together
plotsign.vs.nonSign = rbind(Nonsignificant_df, Significant_df)

#Plot
library(ggplot2)
plotme = ggplot(plotsign.vs.nonSign,
       aes(x = Genus, y = (..count..)/sum(..count..)), stat = "count") + 
  geom_bar() +
  ylab("Number of genera (%)")+
  scale_y_continuous(labels = scales::percent)+
  xlab("")+
  scale_x_discrete(labels=c("Nonsignificant" = "Non-significant", "Significant" = "Significant"))

plotme

dev.copy(png, "Sign.vs.NonSign", units = "in", width = 5, height = 4, res = 500)
dev.off()

#How many significant vs. non-significant?
summary(as.factor(plotsign.vs.nonSign$Genus))

#Total number of sample
 74863       +     29240
#104103

#Ratio of significant
29240/104103
#28.1%

#Ratio of non-significant
74863/104103
#71.9%

```
##Export for ggarrange
```{r Export consensus}
saveRDS(Summary_all_plot,"~/Desktop/Phylum_EdgeR_Consensus_16S.rds")

```



